### DISCUSSION TITLE
psr/container 1^ || ^2

### DISCUSSION THREAD

**Discussion by fezfez** [thecodingmachine/graphqlite#601] [Ideas]
Hello,

I'm stuck with the version 6.0 due to my DI container  `laminas/laminas-servicemanager`, it only support `psr/container:^1.0`.

My idea is to create a sub package of `thecodingmachine/graphqlite` named `thecodingmachine/graphqlite-container` and move the namespace `TheCodingMachine\GraphQLite\Containers` in it,  implementation of `psr/container:^1.0`  tagged as 1.x and `psr/container:^2.0` tagged as 2.x.

This way composer will resolve the good container implementation itself.

WDYT ?

Thanks !

**Comment by oojacoboo** [comment]
Why not just get `laminas/laminas-servicemanager` to update for 2.0 support?

**Reply by fezfez** [reply]
V4 will support it but it's in Dec for 1 year now, heavy dev not that's simple

**Comment by oojacoboo** [comment]
Sigh... I'm not a huge fan of adding this complexity, especially for BC support.  Will GraphQLite even be able to handle both of these container implementations?  I haven't looked into how containers are being called within GraphQLite and what the reliance is.

**Reply by fezfez** [reply]
there is no bc in the caller, only the implementation

**Comment by oojacoboo** [comment]
Okay, feel free to submit a PR on it.  Let's stick with a locked version in composer, please.
